docs/upgrading.md

```diff
@@ -1,66 +1,104 @@
+Currently, you can enable leader election with the `--enable-leader-election` flag to
+`contour serve`. This will ensure that *only the elected leader serves the gRPC endpoints.*
+That is, only the elected leader will actually open the gRPC port and start serving.
+
+If you have deployed Contour and Envoy in their own pods - we call this split deployment - you
+should enable leader election so all envoy pods take their configuration from the lead contour. 
 
-If you are running using the `ds-hostnet-split` example or a derivative, we strongly recommend that you generate new certificates for securing your gRPC communication between Contour and Envoy.
+There are some further requirements for this, see the [howto](#enabling-leader-election) section below for more.
 
-There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
+### Enabling TLS for gRPC
+
+There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command
+to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly
+to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
 
 Again, check the examples for Contour and Envoy in `ds-hostnet-split` for the exact commands.
 
-The steps you will need to follow for this to work are detailed more in (grpc-tls-howto.md)[./grpc-tls-howto.md]:
+If you wish to do this manually, you can use the steps detailed in (grpc-tls-howto.md)[./grpc-tls-howto.md].
+
+In short, you will need to:
 1. Generate a CA keypair
-1. Use it to generate and sign two more keypairs, one for Contour and one for Envoy.
-1. Get the CA certificate (only) and the Contour and Envoy keypairs into Kubernetes Secrets inside the `heptio-contour` (or whatever) namespace.
-1. Mount the CA cert and the keypair into the Contour and Envoy pods, and supply the correct parameters to `contour serve` (for Contour) and `contour bootstrap` (for Envoy)
+2. Use it to generate and sign two more keypairs, one for Contour and one for Envoy.
+3. Get the CA certificate (only) and the Contour and Envoy keypairs into Kubernetes Secrets inside the `heptio-contour` (or whatever) namespace.
+5. Mount the CA cert and the keypair into the Contour and Envoy pods, and supply the correct parameters to `contour serve` (for Contour) and `contour bootstrap` (for Envoy)
+
+As mentioned in (grpc-tls-howto.md)[./grpc-tls-howto.md], the `contour certgen` command can
+perform some or all of the changes for you.
 
-If you are interested in the exact details of how the certificates need to be generated, see `docs/grpc-tls-howto.md`, which explains the `openssl` commands required to perform the same steps as `contour certgen` can do for you.
+### Enabling Leader Election
```

Comment:
Please add an explanatory paragraph about why the reader cares about leader election. That way they can decide to skip this section or not.